src/components/Greeting.js

```diff
@@ -0,0 +1,28 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { fetchRandomGreeting } from '../redux/actions';
+
+const Greeting = ({ greeting, fetchRandomGreeting }) => {
+  useEffect(() => {
+    fetchRandomGreeting();
+  }, [fetchRandomGreeting]);
+
+  return (
+    <div>
+      <h1>Random Greeting</h1>
+      <p>{greeting}</p>
+    </div>
+  );
+};
+
+Greeting.propTypes = {
+  greeting: PropTypes.string.isRequired,
+  fetchRandomGreeting: PropTypes.func.isRequired,
+};
+
+const mapStateToProps = (state) => ({
+  greeting: state.greeting,
+});
+
+export default connect(mapStateToProps, { fetchRandomGreeting })(Greeting);
```

Comment:
**OPTIONAL** - You have done well so far, but since you are working with functional components and not class components, it is advised to use the `useSelector` hook to fetch the app's state from redux. Please fix this 🙏 
**OPTIONAL** - Also, please make use of the `useDispatch` hook to dispatch your API calls.